src/main/java/com/google/firebase/auth/FirebaseTokenVerifierImpl.java

```diff
@@ -96,7 +97,9 @@ private FirebaseTokenVerifierImpl(Builder builder) {
   public FirebaseToken verifyToken(String token) throws FirebaseAuthException {
     IdToken idToken = parse(token);
     checkContents(idToken);
-    checkSignature(idToken);
+    if (!isEmulatorMode()) {
```

Comment:
Personally I prefer the ability to programmatically enable the emulator mode, instead of just tying it to an environment variable. That makes unit testing a lot easier. However, this is the approach we've taken in our Node.js SDK so we can live with this.

But let's make sure we only lookup the env variable once per token verification. So assign the result to a local variable, and use it in rest of the method body:

```
boolean emulator = Utils.isEmulatorMode();
```

Reply:
Done.